### ISSUE TITLE
Crash neeeddd gegelo

### ISSUE THREAD

**Issue by akjkmeagabase** [Tobi823/ffupdater#349]
Stacktrace:
```
de.marmaro.krt.ffupdater.network.exceptions.NetworkException: Download of https://github.com/bromite/bromite/releases/download/108.0.5359.156/arm_ChromePublic.apk failed.
	at de.marmaro.krt.ffupdater.network.FileDownloader$downloadBigFileAsync$deferred$1.invokeSuspend(SourceFile:166)
	at z3.a.resumeWith(Unknown Source:8)
	at l4.l0.run(Unknown Source:100)
	at p4.e.run(Unknown Source:11)
	at r4.j.run(Unknown Source:2)
	at r4.a$a.run(SourceFile:82)
Caused by: java.net.SocketTimeoutException: timeout
	at c5.r$c.j(Unknown Source:4)
	at c5.r$c.l(Unknown Source:8)
	at c5.r$b.read(SourceFile:137)
	at z4.c$b.read(SourceFile:16)
	at h5.r.read(SourceFile:4)
	at h5.i.read(Unknown Source:7)
	at de.marmaro.krt.ffupdater.network.ProgressResponseBody$trackTransmittedBytes$1.read(Unknown Source:5)
	at h5.r$a.read(SourceFile:4)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:347)
	at java.io.FilterInputStream.read(FilterInputStream.java:107)
	at androidx.activity.m.k(Unknown Source:9)
	at de.marmaro.krt.ffupdater.network.FileDownloader.downloadBigFileInternal(Unknown Source:135)
	at de.marmaro.krt.ffupdater.network.FileDownloader.access$downloadBigFileInternal(Unknown Source:0)
	at de.marmaro.krt.ffupdater.network.FileDownloader$downloadBigFileInternal$1.invokeSuspend(Unknown Source:12)
	... 5 more
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.6.3 (125) release |
| Device | S6303L (A9LP_S, S6303LA_Go, NUU) |
| Manufacturer | NUU (NUU) | 
| Supported ABIs | armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | AGENEW, lxc, release-keys, 1668392395000 |         
``

**Comment by ryder203** [comment]
Stacktrace:
```
java.lang.IllegalArgumentException: Failed requirement.
	at de.marmaro.krt.ffupdater.installer.ApkChecker$Companion.throwIfApkFileIsNoValidZipFile(ApkChecker.kt:17)
	at de.marmaro.krt.ffupdater.DownloadActivity.startDownload(DownloadActivity.kt:330)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$startDownload(DownloadActivity.kt:58)
	at de.marmaro.krt.ffupdater.DownloadActivity$startDownload$1.invokeSuspend(DownloadActivity.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7872)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@137d0e3, Dispatchers.Main]
Timestamp: 2023-02-27T12:06:06.736

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.6 (132) release |
| Device | Pixel 7 (panther, panther, panther) |
| Manufacturer | google (Google) | 
| Supported ABIs | arm64-v8a |
| Android version | 13 (SDK: 33) |
| OS | abfarm-release-2004-0153, android-build, release-keys, 1673310668000 |

**Comment by Tobi823** [comment]
@ryder203 Your bug is fixed in 77.7.8

@akjkmeagabase I think that your smartphone just lost the internet connection and the download failed. Nothing special if this only occurs only rarely. If it occurs often, please open a new issue with the new error message.
